dipy/tracking/local_tracking.py

```diff
@@ -129,45 +157,56 @@ def _generate_tractogram(self):
 
         F = np.empty((self.max_length + 1, 3), dtype=float)
         B = F.copy()
-        for s in self.seeds:
+        for i, s in enumerate(self.seeds):
             s = np.dot(lin, s) + offset
-            # Set the random seed in numpy, random and fast_numpy (libc.stdlib)
+            # Set the random seed in numpy, random and fast_numpy (lic.stdlib)
             if self.random_seed is not None:
                 s_random_seed = hash(np.abs((np.sum(s)) + self.random_seed)) \
                     % (2**32 - 1)
                 random.seed(s_random_seed)
                 np.random.seed(s_random_seed)
                 fast_numpy.seed(s_random_seed)
-            directions = self.direction_getter.initial_direction(s)
-            if directions.size == 0 and self.return_all:
+
+            if self.initial_directions is None:
+                directions = self.direction_getter.initial_direction(s)
+            else:
+                d_ns = np.linalg.norm(self.initial_directions[i, :, :], axis=1)
+                directions = self.initial_directions[i, d_ns > 0, :] \
+                    / d_ns[d_ns > 0, np.newaxis]
```

Comment:
not sure to understand what you are doing here. Can you explain ? thank you

Reply:
I added a comment in the code. On line 173, I compute the norm of the initial directions. Here, I normalized all directions that have a norm > 0, and remove initial directions that have `norm==0`.